plugins/package-managers/bazel/src/main/kotlin/StarlarkParser.kt

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.bazel
+
+data class ModuleMetadata(
+    val module: ModuleDirective? = null,
+    val dependencies: List<BazelDepDirective>
+)
+
+data class ModuleDirective(
+    val name: String,
+    val version: String,
+    val compatibilityLevel: Int
+)
+
+data class BazelDepDirective(
+    val name: String,
+    val version: String,
+    val devDependency: Boolean
+)
+
+enum class TokenType {
+    IDENTIFIER, STRING, NUMBER, BOOLEAN, EQUALS, COMMA, LPAREN, RPAREN, LBRACKET, RBRACKET, EOF
+}
+
+data class Token(val type: TokenType, val value: String)
+
+class Lexer(private val input: String) {
```

Comment:
Is this code generated or manually written?

Reply:
It's generated.

Reply:
Shall we add a KDoc comment which states how it was generated? 

Is there any particular license which applies to the generated code?

Reply:
The code was [generated by GPT-4](https://chat.openai.com/share/6cc4f4e1-9ac8-4eb7-8dd0-fd8476a38537). Should I add the link to that chat log to the code? Does ORT have a policy regarding code generated by an LLM?

Reply:
> Does ORT have a policy regarding code generated by an LLM?

Argh, I wish you wouldn't have asked...

Reply:
Shall we just add a comment to either the class or the file, which indicates that / how the file was generated?

Reply:
I've added a comment to the top of the file.